src/fermilib/circuits/ffft.py

```diff
@@ -0,0 +1,383 @@
+"""Functions to perform the fermionic fast Fourier transform."""
+import itertools
+import numpy
+
+from fermilib.circuits._parallel_bubble_sort import (
+    index_of_position_in_1d_array, parallel_bubble_sort)
+from fermilib.ops import FermionOperator, normal_ordered
+from fermilib.transforms import jordan_wigner
+from fermilib.utils import fourier_transform, Grid
+
+from projectq import MainEngine
+from projectq.ops import (QubitOperator, H, X, Y, Z, C, Rx, Ry, Rz, Ph, Swap,
+                          Measure, All, TimeEvolution)
+
+
+def fswap_generator(mode_a, mode_b):
+    """Give the two-qubit fermionic swap operator between two modes.
+
+    Args:
+        mode_a, mode_b (int): The two modes to be swapped.
+
+    Returns:
+        FermionOperator which swaps modes mode_a and mode_b.
+
+    Notes:
+        A fermionic swap is the same as the usual swap, with the
+        difference that it applies a sign -1 to the state |11>.
+    """
+    return (FermionOperator('') +
+            FermionOperator(((mode_a, 1), (mode_b, 0))) +
+            FermionOperator(((mode_b, 1), (mode_a, 0))) -
+            FermionOperator(((mode_a, 1), (mode_a, 0))) -
+            FermionOperator(((mode_b, 1), (mode_b, 0))))
+
+
+def fswap(register, mode_a, mode_b):
```

Comment:
Since this is a general gate, don't you think there should be an optional parameter that says "use_jordan_wigner=true" or something. And then you use BK otherwise?